File: week11/19/P1.py
class Solution(object):
    def isPalindrome(self, s):
        c = ""
        for ch in s:
            if ch.isalnum():
                c += ch.lower()

        def lpsHelp(s, start, end):
            if (end - start < 0):
                return False
            if (s[start] != s[end]):
                return False
            if (0 <= end - start <= 1):
                return True
            return lpsHelp(s, start+1, end-1)
        
        if (c == ""):
            return True
        return lpsHelp(c, 0, len(c)-1)

if __name__ == "__main__":
    result = Solution()
    print(result.isPalindrome("A man, a plan, a canal: Panama"))
    print(result.isPalindrome("race a car"))
    print(result.isPalindrome(" "))

File: week11/19/P1.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        int start=0;
        int end=s.size()-1;
        while(start<=end){
            if(!isalnum(s[start])){start++; continue;}
            if(!isalnum(s[end])){end--;continue;}
            if(tolower(s[start])!=tolower(s[end]))return false;
            else{
                start++;
                end--;
            }
        }
        return true;
    }   
};

int main() {
    Solution result;
    cout << result.isPalindrome("A man, a plan, a canal: Panama") << endl;
    cout << result.isPalindrome("race a car") << endl;
    cout << result.isPalindrome(" ") << endl;

    return 0;
}